### ISSUE TITLE
@types/node-forge - Property 'encrypt' does not exist on type 'PublicKey'

### ISSUE THREAD

**Issue by umdstu** [DefinitelyTyped/DefinitelyTyped#38541]
If you know how to fix the issue, make a pull request instead.

- [x ] I tried using the `@types/xxxx` package and had problems.
- [x ] I tried using the latest stable version of tsc. https://www.npmjs.com/package/typescript
- [x ] I have a question that is inappropriate for [StackOverflow](https://stackoverflow.com/).  

@westy92 @Apologiz @flynetworks  @Rafal2228 @beenotung @joeflateau @timhwang21 @supaiku0 @andersk @saschazar21 @a-k-g 

There is an [issue ](https://github.com/digitalbazaar/forge/issues/664) for this in the node-forge` package itself, but suspect it may be related to the type definitions.

```
const publicKey = forge.pki.publicKeyFromPem(publicKeyString);
const text = forge.util.hexToBytes('words to encrypt');
publicKey.encrypt(text, 'RSA-OAEP');
```

vscode error:
```
any
Property 'encrypt' does not exist on type 'PublicKey'.
    Property 'encrypt' does not exist on type 'Buffer'. ts(2339)
```

Hoping it's obvious to you what might need to be changed in the type definitions for this to resolve.

Thanks in advance.

**Comment by umdstu** [comment]
Update:

It looks like removing the `or` for the PublicKey type in the `pki` namespace fixes the issue, otherwise it's cast as an `ed25519.Key`

I'd do a PR but I have no clue what impacts doing this might have.

TLDR;
`forge.pki.publicKeyFromPem(pem)` returns a `forge.pki.PublicKey` but if you remove `| ed25519.Key` from the `pki` namespace for `type PublicKey`, `forge.pki.publicKeyFromPem(pem)` instead returns a `forge.pki.rsa.PublicKey` and the `encrypt` and `verify` functions are picked up.

**Comment by saschazar21** [comment]
Hi @umdstu - thanks for your documentation. Please feel free to create a PR, because it needs a review anyhow, therefore I think it's unlikely to 'break stuff' entirely without anybody else knowing.

If you need assistance anywhere, please continue commenting below.

**Comment by josemsantos** [comment]
Hey, I'm having the same issue. 
`Forge.pki.publicKeyFromAsn1(subjectPublicKeyInfo)`
returns a Forge.pki.Public while it should be Forge.pki.rsa.Public which makes 
`publicKey.encrypt(data, 'RSA-OAEP', {...})`
Incompatible

To go around this issue I've cast the result publicKeyFromAsn1 of it like this
`let publicKey = Forge.pki.publicKeyFromAsn1(subjectPublicKeyInfo) as Forge.pki.rsa.PublicKey`

There's clearly a namespace issue somewhere.

**Comment by Frangello** [comment]
@umdstu did you ever open a PR for this issue?

**Comment by umdstu** [comment]
I did not, sorry. I person i was helping moved away and so i wasn’t using
this package any longer.
